src/PrestaShopBundle/Translation/Provider/TranslationFinder.php

```diff
@@ -27,13 +27,132 @@
 
 namespace PrestaShopBundle\Translation\Provider;
 
-use PrestaShop\PrestaShop\Core\Translation\Finder\TranslationFinder as CoreTranslationFinder;
+use PrestaShop\PrestaShop\Core\Exception\FileNotFoundException;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+use Symfony\Component\Translation\Loader\XliffFileLoader;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\MessageCatalogueInterface;
 
 /**
  * Helper used to build a MessageCataloguer from xliff files
  *
  * @deprecated Please use PrestaShop\PrestaShop\Core\Translation\Provider\TranslationFinder instead
  */
-class TranslationFinder extends CoreTranslationFinder
+class TranslationFinder
 {
+    private const ERR_NO_FILES_IN_DIRECTORY = 1;
+    private const ERR_DIRECTORY_NOT_FOUND = 2;
+
+    /**
+     * @param array $paths a list of paths when we can look for translations
+     * @param string $locale the Symfony (not the PrestaShop one) locale
+     * @param string|null $pattern a regular expression
+     *
+     * @return MessageCatalogue
+     *
+     * @throws FileNotFoundException
+     */
+    public function getCatalogueFromPaths(array $paths, string $locale, string $pattern = null): MessageCatalogue
+    {
+        $translationFiles = $this->getTranslationFilesFromPath($paths, $pattern);
+
+        return $this->buildCatalogueFromFiles($translationFiles, $locale);
+    }
+
+    /**
+     * @param MessageCatalogueInterface $catalogue
+     *
+     * @return MessageCatalogue
+     */
+    private function removeTrailingLocaleFromDomains(MessageCatalogueInterface $catalogue): MessageCatalogue
+    {
+        $messages = $catalogue->all();
+        $locale = $catalogue->getLocale();
+        $localeSuffix = '.' . $locale;
+        $suffixLength = strlen($localeSuffix);
+
+        foreach ($catalogue->getDomains() as $domain) {
+            if (substr($domain, -$suffixLength) === $localeSuffix) {
+                $cleanDomain = substr($domain, 0, -$suffixLength);
+                $messages[$cleanDomain] = $messages[$domain];
+                unset($messages[$domain]);
+            }
+        }
+
+        return new MessageCatalogue($locale, $messages);
+    }
+
+    /**
+     * @param string[] $paths
+     * @param string $pattern
+     *
+     * @return Finder
+     *
+     * @throws FileNotFoundException
+     */
+    private function getTranslationFilesFromPath(array $paths, string $pattern): Finder
+    {
+        $finder = new Finder();
+
+        if (null !== $pattern) {
+            $finder->name($pattern);
+        }
+
+        try {
+            $translationFiles = $finder->files()->notName('index.php')->in($paths);
+        } catch (\InvalidArgumentException $e) {
```

Comment:
```suggestion
        } catch (\InvalidArgumentException $e) {
```

Please move class to use section